mm/2s2h/Enhancements/Fixes/CompletedHeartContainerAudio.cpp

```diff
@@ -0,0 +1,16 @@
+#include <libultraship/bridge.h>
+#include "2s2h/GameInteractor/GameInteractor.h"
+
+extern "C" {
+#include "variables.h"
+}
+
+void RegisterCompletedHeartContainerAudio() {
+    REGISTER_VB_SHOULD(VB_COMPLETED_HEART_CONTAINER_AUDIO_FIX, {
+        GetItemId getItemId = va_arg(args, GetItemId);
```

Comment:
With `va_arg()` you have to be careful what type you specify. The va_arg macro is sensitive to the size of the type passed in.
The first thing of note is that enum types like `GetItemId` should not be used. This is because not all platforms assign enums with the same "size". Usually enums will be 32bit, but Linux will make them smaller.
The second thing is that arguments passed into a variadic function will go through a "promotion" process. Specifically, smaller types like `bool`, `char`, `short` will be promoted to `int`.

As it is right now, this line will crash on Linux.

The `getItemId` from the Player is a `s16` (`short`), so you will need to parse it as a `s32`/`int` on the va_arg side. Then you can cast it into a `GetItemId`.

```suggestion
        GetItemId getItemId = (GetItemId)va_arg(args, int);
```

Reply:
Thank you. When I get the time, I will read more into variadic functions. I have made your suggested change. But I've been unable to test non-Windows versions of 2ship as I don't have setups for it.